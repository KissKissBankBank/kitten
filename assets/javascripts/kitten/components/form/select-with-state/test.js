import React from 'react'
import renderer from 'react-test-renderer'
import { SelectWithState } from './index'

describe('<SelectWithState />', () => {
  let select
  const options = [
    { value: 'foo', label: 'Foo' },
    { value: 'bar', label: 'Bar' },
  ]
  const defaultValue = { value: 'bar', label: 'Bar' }

  describe('with default props', () => {
    beforeAll(() => {
      select = renderer.create(<SelectWithState options={options} />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(select).toMatchSnapshot()
    })
  })

  describe('with default value', () => {
    beforeAll(() => {
      select = renderer
        .create(<SelectWithState options={options} value={defaultValue} />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(select).toMatchSnapshot()
    })
  })

  describe('with label', () => {
    beforeAll(() => {
      select = renderer
        .create(<SelectWithState options={options} labelText="FooBar" />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(select).toMatchSnapshot()
    })
  })

  describe('with classes props', () => {
    beforeAll(() => {
      select = renderer
        .create(<SelectWithState options={options} tiny error valid disabled />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(select).toMatchSnapshot()
    })
  })
})
