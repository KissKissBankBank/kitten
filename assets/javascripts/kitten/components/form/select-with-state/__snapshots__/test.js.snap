// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectWithState /> with classes props matches with snapshot 1`] = `
<div
  className="k-Select k-Select--tiny is-error is-valid is-disabled"
>
  <div
    className="Select Select--single is-disabled has-value"
  >
    <div
      className="Select-control"
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <span
        className="Select-multi-value-wrapper"
        id="react-select-5--value"
      >
        <div
          className="Select-value"
        >
          <span
            aria-selected="true"
            className="Select-value-label"
            id="react-select-5--value-item"
            role="option"
          />
        </div>
        <div
          aria-activedescendant="react-select-5--value"
          aria-disabled="true"
          aria-expanded={false}
          aria-owns=""
          className="Select-input"
          onBlur={[Function]}
          onFocus={[Function]}
          role="combobox"
          style={
            Object {
              "border": 0,
              "display": "inline-block",
              "width": 1,
            }
          }
          tabIndex={0}
        />
      </span>
      <span
        className="Select-arrow-zone"
        onMouseDown={[Function]}
      >
        <span
          className="Select-arrow"
          onMouseDown={[Function]}
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`<SelectWithState /> with default props matches with snapshot 1`] = `
<div
  className="k-Select"
>
  <div
    className="Select Select--single has-value"
  >
    <div
      className="Select-control"
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <span
        className="Select-multi-value-wrapper"
        id="react-select-2--value"
      >
        <div
          className="Select-value"
        >
          <span
            aria-selected="true"
            className="Select-value-label"
            id="react-select-2--value-item"
            role="option"
          />
        </div>
        <div
          aria-activedescendant="react-select-2--value"
          aria-disabled="false"
          aria-expanded={false}
          aria-owns=""
          className="Select-input"
          onBlur={[Function]}
          onFocus={[Function]}
          role="combobox"
          style={
            Object {
              "border": 0,
              "display": "inline-block",
              "width": 1,
            }
          }
          tabIndex={0}
        />
      </span>
      <span
        className="Select-arrow-zone"
        onMouseDown={[Function]}
      >
        <span
          className="Select-arrow"
          onMouseDown={[Function]}
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`<SelectWithState /> with default value matches with snapshot 1`] = `
<div
  className="k-Select"
>
  <div
    className="Select Select--single has-value"
  >
    <div
      className="Select-control"
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <span
        className="Select-multi-value-wrapper"
        id="react-select-3--value"
      >
        <div
          className="Select-value"
        >
          <span
            aria-selected="true"
            className="Select-value-label"
            id="react-select-3--value-item"
            role="option"
          />
        </div>
        <div
          aria-activedescendant="react-select-3--value"
          aria-disabled="false"
          aria-expanded={false}
          aria-owns=""
          className="Select-input"
          onBlur={[Function]}
          onFocus={[Function]}
          role="combobox"
          style={
            Object {
              "border": 0,
              "display": "inline-block",
              "width": 1,
            }
          }
          tabIndex={0}
        />
      </span>
      <span
        className="Select-arrow-zone"
        onMouseDown={[Function]}
      >
        <span
          className="Select-arrow"
          onMouseDown={[Function]}
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`<SelectWithState /> with label matches with snapshot 1`] = `
<div
  className="k-Select"
>
  <label
    className="k-Select__label"
  >
    FooBar
  </label>
  <div
    className="Select Select--single has-value"
  >
    <div
      className="Select-control"
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <span
        className="Select-multi-value-wrapper"
        id="react-select-4--value"
      >
        <div
          className="Select-value"
        >
          <span
            aria-selected="true"
            className="Select-value-label"
            id="react-select-4--value-item"
            role="option"
          />
        </div>
        <div
          aria-activedescendant="react-select-4--value"
          aria-disabled="false"
          aria-expanded={false}
          aria-owns=""
          className="Select-input"
          onBlur={[Function]}
          onFocus={[Function]}
          role="combobox"
          style={
            Object {
              "border": 0,
              "display": "inline-block",
              "width": 1,
            }
          }
          tabIndex={0}
        />
      </span>
      <span
        className="Select-arrow-zone"
        onMouseDown={[Function]}
      >
        <span
          className="Select-arrow"
          onMouseDown={[Function]}
        />
      </span>
    </div>
  </div>
</div>
`;
